src/opdcodes.rs

```diff
@@ -47,6 +50,119 @@ impl From<u16> for Opcode {
     }
 }
 
+// TODO: improve error handling
+pub fn trap(vm: &mut VM, instruction: u16) -> Result<(), VMError> {
+    vm.write_register(7, vm.registers.pc);
+
+    let trap_vector = instruction & 0xFF;
+
+    match trap_vector {
+        0x20 => {
+            // GETC - Read a single character from the keyboard, The character is not echoed onto the console.
+            // Its ASCII code is copied into register 0. The high 8 bits of R0 are cleared.
+            let mut buffer = [0; 1];
+            std::io::stdin()
+                .read_exact(&mut buffer)
+                .map_err(|err| VMError::TrapError(TrapError::IOError(err.to_string())))?;
+
+            if let Some(c) = buffer.first() {
+                vm.registers.set(0, (*c).into());
+                vm.update_flags(0);
+            }
+            Ok(())
+        }
```

Comment:
shouldn't it be better to wrap every block of the match with a function call?

Reply:
I wanted to avoid wrapping every trap handle in a function or creating abstracion for this. What do you think?

Reply:
I think it's a matter of personal taste :D

Reply:
I think I agree with Tomi here; like he said, it's just personal taste. However, I think it makes the program a bit clearer to have a function that describe _what_ it will do, rather than _how_ it will do it.